jcl/src/java.base/share/classes/jdk/internal/foreign/abi/UpcallLinker.java

```diff
@@ -113,5 +121,24 @@ public static MemorySegment make(MethodHandle target, MethodType mt, FunctionDes
 	public static UpcallStubFactory makeFactory(MethodType targetType, ABIDescriptor abi, CallingSequence callingSequence) {
 		throw new UnsupportedOperationException();
 	}
+
+	/**
+	 * The method constructs a factory instance (introduced in JDK21 to support the
+	 * sharing of function descriptor) which wraps up the existing upcall specific
+	 * code implemented in OpenJ9 to enable the callback in OpenJDK.
+	 *
+	 * @param targetType
+	 * @param abi
+	 * @param callingSequence
+	 * @return
+	 * @throws IllegalArgumentException
```

Comment:
Please use clearer parameter names (e.g. mt -> methodType, cDesc -> descriptor) and correct the javadoc so the `@param` elements each match an actual parameter and describe what they're about.

Reply:
Updated the `@param` elements with descriptions as suggested above.